CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+
+The instrumentation tests presume the presence of various backend resources.
+Currently, there is no mechanism for contributors to easily allocate
+these resources. 
+
+AWS maintainers can gain access to `awsconfiguration.json` and
+`testconfiguration.json` files in an S3 bucket, to find
+configurations suitable for running the integration tests.
+
+The file locations can be found in the `Configure test information` command in the `.circleci/config.yml` file.
+
+To run a __specific__ test:
+
+```bash
+test='com.amazonaws.mobileconnectors.cognitoidentityprovider.SignInPersistenceIntegrationTest#testSignIn'
+./gradlew cAT -Pandroid.testInstrumentationRunnerArguments.class="$test"
+```
+
+To run __all__ tests:
+
+```
+./gradlew cAT
+```
+
+## Reporting Bugs/Feature Requests
+
+We welcome you to use the GitHub issue tracker to report bugs or suggest
+features.
+
+When filing an issue, please check [existing open](https://github.com/aws-amplify/aws-sdk-android/issues)
+and [recently closed](https://github.com/aws-amplify/aws-sdk-android/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20)
+issues to make sure somebody else hasn't already reported the issue.
+Please try to include as much information as you can. Details like these
+are useful:
+
+* The version of the Framework you are using
+* Details and configurations for any backend resources that are relevant
+* A full exception trace of an error you observe
+* A statement about what system behavior you _expect_, alongside the
+  behavior you actually observe
+
+## Contributing via Pull Requests
+
+This is mostly the same as [GitHub's guide on creating a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).
+
+First, create a _fork_ of `aws-sdk-android`. Clone it, and make changes to this _fork_.
+
+```
+git clone git@github.com:your_username/aws-sdk-android.git 
+```
+
+After you have tested your feature/fix, by adding sufficient test coverage, and
+validating Checkstyle, lint, and the existing test suites, you're ready to
+publish your change.
+
+The commit message should look like below. It started with a bracketed tag
+stating which module has been the focus of the change. After a paragraph
+describing what you've done, include links to useful resources. These might
+include design documents, StackOverflow implementation notes, GitHub issues,
+etc. All links must be publicly accessible.
+
+__NOTE__: this is not an actual change that has been implemented. Only an example commit message.
+
+```
+[aws-android-sdk-core] Added new pre-defined retry policy
+
+Added a new retry policy to allow retries on select exception types
+
+Resolves: https://github.com/aws-sdk-android/issues/XXXX
+```
+
+Now, save your work to a new branch:
+
+```
+git checkout -B new_retry_policy
+```
+
+To publish it:
+
+```
+git push origin origin new_retry_policy
+```
+
+This last step will give you a URL to view a GitHub page in your browser.
```

Comment:
Wording is awkward, maybe -- `This last step will output an GitHub URL, that you can view in your web browser.`